packages/request-client.js/src/api/request-network.ts

```diff
@@ -66,6 +71,81 @@ export default class RequestNetwork {
     );
   }
 
+  /**
+   * Prepares a payment request structure from transaction data.
+   *
+   * This method is used to create a request structure similar to a persisted request,
+   * allowing users to pay before the request is persisted. This is useful in scenarios
+   * where a request is created, paid, and then persisted, as opposed to the normal flow
+   * of creating, persisting, and then paying the request.
+   *
+   * @param transactionData The transaction data containing the request information
+   * @param requestId The ID of the request
+   * @returns The prepared payment request structure or undefined if transaction data is missing
+   */
+  private preparePaymentRequest(
+    transactionData: DataAccessTypes.ITransaction,
+    requestId: string,
+  ): ClientTypes.IRequestData | undefined {
+    if (!transactionData.data) return undefined;
+
+    const requestData = JSON.parse(transactionData.data).data;
```

Comment:
This kind of `data.data.data` looks super suspicious but I did not check. Couldn't you rely on something more precise?

Reply:
I heard @aimensahnoun say he agreed, but due to the types, it was unavoidable.